File: db-python-ex/example/DB_01_ConnectOracle.py
import cx_Oracle

cx_Oracle.init_oracle_client(r"C:\Users\leeyoungseung\project_source\database-ex\instantclient_11_2")

# Oracle 데이터베이스에 접속
dsn = cx_Oracle.makedsn(host='localhost', port='1521', sid='xe')
connection = cx_Oracle.connect(user='hr', password='1234', dsn=dsn)

# 커넥션 상태 확인
if connection:
    print("Oracle DB 연결 성공")
else:
    print("Oracle DB 연결 실패")
    exit(100)

# 접속된 데이터베이스에서 쿼리 실행
cursor = connection.cursor()
cursor.execute("SELECT * FROM JOBS WHERE ROWNUM <= 100")
result = cursor.fetchall()

# 쿼리 결과 출력
for row in result:
    print(row)

# 연결 종료
connection.close()
exit(0)

File: db-perl-ex/src/DB_01_ConnectOracle.pl
## Perl을 사용해한 오라클접속은 결국 시간내에 달성하지 못함.
# strawberry perl 환경에서 두개 라이브러리를 사용해 봤지만 실패했음.
# DBD::Oracle : 애초에 cpan으로 설치가 안됨. 전제조건인 오라클 클라이언트 설치 및 환경변수 설정은 제대로 되었지만 안됨...노답 
# DBD::ODBC   : 설치는 됬지만 더 투자할 시간이 아까워 디질것 같음. 

use DBI;

# 데이터베이스 연결 설정
my $hostname = "localhost";
my $port = 5432;
my $username = "hr";
my $password = "1521";
my $database = "xe";

# Oracle 데이터베이스에 접속
my $dsn = "dbi:ODBC:DSN=$hostname:$port:";
my $dbh = DBI->connect($dsn, $username, $password);

if ($dbh) {
    print "Database connection success \n";
} else {
    print "Database connection error: $DBI::errstr\n";
    exit;
}

$dbh->disconnect();